CI: Test/TunnelMapTestSuite/TestClusterAwareAddressing

## CI failure

```
--- FAIL: Test (0.00s)
    --- FAIL: Test/TunnelMapTestSuite (0.00s)
        --- FAIL: Test/TunnelMapTestSuite/TestClusterAwareAddressing (0.00s)
            tunnel_test.go:37: 
                ... value *fmt.wrapError = &fmt.wrapError{msg:"creating map: pin map to /sys/fs/bpf/tc/globals/test_cilium_tunnel_map: file exists", err:(*fmt.wrapError)(0xc0000563c0)} ("creating map: pin map to /sys/fs/bpf/tc/globals/test_cilium_tunnel_map: file exists")
```

Run URL: https://github.com/cilium/cilium/actions/runs/8193113970/job/22406307012
PR: https://github.com/cilium/cilium/pull/31015
Zip: Sysdump n/a.

--- a/pkg/maps/tunnel/tunnel.go
+++ b/pkg/maps/tunnel/tunnel.go
@@ -33,6 +33,10 @@ var (
 
 // SetTunnelMap sets the tunnel map. Only used for testing.
 func SetTunnelMap(m *Map) {
+	if tunnelMap != nil {
+		tunnelMap.UnpinIfExists()
+	}
+
 	tunnelMap = m
 }
 

--- a/pkg/maps/tunnel/tunnel.go
+++ b/pkg/maps/tunnel/tunnel.go
@@ -42,7 +42,9 @@ func SetTunnelMap(m *Map) {
 
 func TunnelMap() *Map {
 	tunnelMapInit.Do(func() {
-		tunnelMap = NewTunnelMap(MapName)
+		if tunnelMap == nil {
+			tunnelMap = NewTunnelMap(MapName)
+		}
 	})
 	return tunnelMap
 }

--- a/pkg/maps/tunnel/tunnel_test.go
+++ b/pkg/maps/tunnel/tunnel_test.go
@@ -23,7 +23,6 @@ func setupTunnelMapTestSuite(tb testing.TB) {
 func TestClusterAwareAddressing(t *testing.T) {
 	setupTunnelMapTestSuite(t)
 	m := NewTunnelMap("test_cilium_tunnel_map")
-	defer m.Unpin()
 
 	err := m.OpenOrCreate()
 	require.Nil(t, err)
@@ -66,4 +65,7 @@ func TestClusterAwareAddressing(t *testing.T) {
 
 	_, err = m.GetTunnelEndpoint(prefix1)
 	require.Error(t, err)
+
+	err = m.Unpin()
+	require.Nil(t, err)
 }

